src/main/java/io/cryostat/jfr/datasource/server/JfrResource.java

```diff
@@ -126,6 +126,31 @@ void list(RoutingContext context) {
         response.end(responseBuilder.toString());
     }
 
+    @Route(path = "/delete_all")
+    @Blocking
+    void deleteAll(RoutingContext context) {
+        HttpServerResponse response = context.response();
+        setHeaders(response);
+
+        final StringBuilder stringBuilder = new StringBuilder();
```

Comment:
This calling convention of passing in a `StringBuilder`  instance looks a little funny to me, like it's an "output parameter".

For deleting a single file I don't think it's necessary to respond back with the file name that was deleted. Simply a `200`/`204` response code seems sufficient.

For deleting all files, I think `deleteAllFiles` could just return `List<String>` of all the files that were deleted, and then this `deleteAll` method can deal with formatting that list and adding the `Deleted: ` before each item or whatever response format makes sense.

Reply:
> This calling convention of passing in a StringBuilder instance looks a little funny to me, like it's an "output parameter".

I was wondering the same :D It seems every handler follows it. What do you think would be a nicer approach?

Reply:
It's not a big deal for now, just something we could probably refactor to improve later.